services/surveys.js

```diff
@@ -0,0 +1,82 @@
+const xss = require('xss');
+const { Survey, SurveyChoice, SurveyResponse } = require('../models');
+
+function findAll() {
+    return Survey.findAll({
+        include: [
+            {
+                model: SurveyChoice,
+                as: 'choices',
+                required: true,
+                include: [
+                    {
+                        model: SurveyResponse,
+                        as: 'responses',
+                        required: true
+                    }
+                ]
+            }
+        ]
+    });
+}
+
+function findActiveWithChoices({ filterByPath }) {
+    return Survey.findAll({
+        where: {
+            active: true
+        },
+        include: [
+            {
+                model: SurveyChoice,
+                as: 'choices',
+                required: true
+            }
+        ]
+    }).then(surveys => {
+        if (filterByPath) {
+            return surveys.filter(s => s.activePath === filterByPath);
+        } else {
+            return surveys;
+        }
+    });
+}
+
+function createWithChoices(data) {
+    return Survey.create(data, {
+        include: [
+            {
+                model: SurveyChoice,
+                as: 'choices'
+            }
+        ]
+    });
+}
+
+function createResponse(submissionBody) {
```

Comment:
Not 100% sure but I think I'd prefer most of this function to stay in the controller and just have the`SurveyResponse.create` functionality live here – think it's a bit confusing having the stuff dealing with form data / validation mixed in with the model stuff (kind of the inverse of the point of this change, right?).

Reply:
Yeah, I think you're right.